src/ServerService.php

```diff
@@ -0,0 +1,140 @@
+use Wsdl2PhpGenerator\PhpSource\PhpDocComment;
+use Wsdl2PhpGenerator\PhpSource\PhpDocElementFactory;
+use Wsdl2PhpGenerator\PhpSource\PhpFunction;
+use Wsdl2PhpGenerator\PhpSource\PhpVariable;
+
+/**
+ * Service represents the service in the wsdl, from the server side
+ *
+ * @package Wsdl2PhpGenerator
+ * @author Thierry Blind <thierryblind@msn.com>
+ * @license http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+class ServerService extends Service
+{
+    const SERVER_SERVICE_PREFIX = 'AbstractServer';
+
+    /**
+     * @var string The class name of the client service
+     */
+    private $clientClassName;
+
+    /**
+     * @param ConfigInterface $config Configuration
+     * @param string $identifier The name of the client service
+     * @param array $types The types the service knows about
+     * @param string $description The description of the service
+     */
+    public function __construct(ConfigInterface $config, $identifier, array $types, $description)
+    {
+        // In most cases, the final class name of the client service is the same than the identifier
+        // of the client service
+        $this->clientClassName = $identifier;
+        $serverIdentifier = $config->get('serverClassName');
+        if (empty($serverIdentifier)) {
+            $serverIdentifier = self::SERVER_SERVICE_PREFIX . $identifier;
+        }
+        parent::__construct($config, $serverIdentifier, $types, $description);
+    }
+
+    /**
+     * Returns the class name of the client service.
+     *
+     * @return string The class name of the client service.
+     */
+    public function getClientClassName()
+    {
+        return $this->clientClassName;
+    }
+
+    /**
+     * Sets the class name of the client service.
+     *
+     * @return string The class name of the client service.
+     */
+    public function setClientClassName($clientClassName)
+    {
+        $this->clientClassName = $clientClassName;
+        $this->class = null;
+    }
+
+    /**
+     * Generates the class if not already generated
+     */
+    public function generateClass()
+    {
+        $serverName = $this->identifier;
+
+        // Generate a valid classname
+        $serverName = Validator::validateClass($serverName, $this->config->get('namespaceName'));
+
+        // uppercase the name
+        $serverName = ucfirst($serverName);
+
+        // Create the class object
+        $comment = new PhpDocComment($this->description);
+        $this->class = new PhpClass($serverName, false, $this->config->get('soapServerClass'), $comment, false, true);
+
+        // Create the constructor
+        $comment = new PhpDocComment();
+        $comment->addParam(PhpDocElementFactory::getParam('array', 'options', 'A array of config values'));
+        $comment->addParam(PhpDocElementFactory::getParam('string', 'wsdl', 'The wsdl file to use'));
+
+        $source = '
+  foreach (' . $this->clientClassName . '::$classmap as $key => $value) {
```

Comment:
Can we avoid referencing the client class here and instead have the classmap duplicated in both server and client or in shared code?